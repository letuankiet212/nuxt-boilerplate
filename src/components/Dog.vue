<template>
  <div class="dog">
    {{ message }}
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  props: {
    repeatNumber: {
      type: Number,
      default: 1,
      required: false
    }
  },
  data() {
    return {
      message: '🐶'.repeat(this.repeatNumber)
    };
  }
});
</script>

<style scoped>
.dog {
  font-size: 10em;
}
</style>
