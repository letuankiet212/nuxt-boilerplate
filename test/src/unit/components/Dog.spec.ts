import { shallowMount } from '@vue/test-utils';
import Dog from '../../../../src/components/Dog.vue';
import dogTestData from './dogTestData';

describe('🐶', () => {
  it('should be displayed 1 times', () => {
    const wrapper = shallowMount(Dog);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should be displayed 3 times', () => {
    const wrapper = shallowMount(Dog, { propsData: { repeatNumber: dogTestData.repeatNumber } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
