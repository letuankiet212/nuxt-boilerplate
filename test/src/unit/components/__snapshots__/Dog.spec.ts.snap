// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🐶 should be displayed 1 times 1`] = `
"<div class=\\"dog\\">
  🐶
</div>"
`;

exports[`🐶 should be displayed 3 times 1`] = `
"<div class=\\"dog\\">
  🐶🐶🐶
</div>"
`;
